virtualizarr/readers/tiff.py

```diff
@@ -1,79 +1,179 @@
-
-        # both group=None and group='' mean to read root group
+    @staticmethod
+    def _construct_manifest_group(
+        store: SupportedStore,
+        path: str,
+        *,
+        group: str | None = None,
+    ) -> ManifestGroup:
+        """
+        Construct a virtual Group from a tiff file.
+        """
+        # TODO: Make an async approach
+        tiff = sync(TIFFVirtualBackend._open_tiff(store=store, path=path))
+        attrs: dict[str, Any] = {}
+        manifest_arrays = {}
         if group:
-            refs = extract_group(refs, group)
+            manifest_arrays[group] = TIFFVirtualBackend._construct_manifest_array(
+                ifd=tiff.ifds[int(group)], path=path
+            )
+        else:
+            for ind, ifd in enumerate(tiff.ifds):
+                manifest_arrays[str(ind)] = (
+                    TIFFVirtualBackend._construct_manifest_array(ifd=ifd, path=path)
+                )
+        return ManifestGroup(arrays=manifest_arrays, attributes=attrs)
 
-        virtual_vars, attrs, coord_names = virtual_vars_and_metadata_from_kerchunk_refs(
-            refs,
-            fs_root=Path.cwd().as_uri(),
-        )
+    @staticmethod
+    def _create_manifest_store(
+        filepath: str,
+        group: str,
+        file_id: str,
+        object_store: SupportedStore,
+    ) -> Store:
+        # TODO: Make this less sketchy, but it's better to use an AsyncTIFF store rather than an obstore store
+        from async_tiff.store import LocalStore as ATStore
 
-        fully_virtual_dataset = construct_fully_virtual_dataset(
-            virtual_vars=virtual_vars,
-            coord_names=coord_names,
-            attrs=attrs,
-        )
+        newargs = object_store.__getnewargs_ex__()
+        at_store = ATStore(*newargs[0], **newargs[1])
 
-        vds = replace_virtual_with_loadable_vars(
-            fully_virtual_dataset,
-            filepath,
-            group=group,
-            loadable_variables=loadable_variables,
-            reader_options=reader_options,
-            indexes=indexes,
-            decode_times=decode_times,
+        # Create a group containing dataset level metadata and all the manifest arrays
+        manifest_group = TIFFVirtualBackend._construct_manifest_group(
+            store=at_store, path=filepath, group=group
         )
+        # Convert to a manifest store
+        return ManifestStore(stores={file_id: object_store}, group=manifest_group)
 
-        return vds.drop_vars(_drop_vars)
+    @staticmethod
+    def open_virtual_dataset(
+        filepath: str,
+        group: str | None = None,
+        drop_variables: Iterable[str] | None = None,
+        loadable_variables: Iterable[str] | None = None,
+        decode_times: bool | None = None,
+        indexes: Mapping[str, xr.Index] | None = None,
+        virtual_backend_kwargs: Optional[dict] = None,
+        reader_options: Optional[dict] = None,
+    ) -> xr.Dataset:
+        raise NotImplementedError
```

Comment:
Once I've merged #522 I think we will be able to enable this reader immediately.